src/display/api.js

```diff
@@ -1742,7 +1751,10 @@ var WorkerTransport = (function WorkerTransportClosure() {
             var size = width * height;
             var rgbaLength = size * 4;
             var buf = new Uint8Array(size * components);
-            var tmpCanvas = createScratchCanvas(width, height);
+            // make sure that this code is not executing in node.js.
+            assert(typeof document !== 'undefined');
+            var canvasFactory = new DOMCanvasFactory();
```

Comment:
Can't we use `this.canvasFactory` (as set above) here? If so, we can remove this line and use `var tmpCanvas = this.canvasFactory.create(width, height);` in the line below.